### PR TITLE
Ptex fixes:

### PR DESCRIPTION
**Pull request by manuelk**
- fix build compiling & linking to accomodate recent code churn in Ptex
- fix FindPTex.cmake module to correctly extract version number
- fix dxPtexViewer & glPtexViewer source to compile with new Ptex namespace changes
- add alpha channel padding function to ptexMipmapLoader as a workaround to the absence of 3-channel DXGI formats
- mirror ptex memory limit function from glPtexViewer to dxPtexViewer

Sorry - the previous PR was messed up : hopefully this will work better...

### CODE REVIEW

File: examples/common/d3d11PtexMipmapTexture.cpp [hunk 1]
```diff
@@ -138,7 +145,7 @@ D3D11PtexMipmapTexture::Create(ID3D11DeviceContext *deviceContext,
             switch (numChannels) {
                 case 1: format = DXGI_FORMAT_R16_FLOAT; break;
                 case 2: format = DXGI_FORMAT_R16G16_FLOAT; break;
-                case 3:assert(false); break;
+                case 3: assert(false); break;
```
Comment (thomthom) [inline examples/common/d3d11PtexMipmapTexture.cpp]: Unintended space?

File: CMakeLists.txt [hunk 2]
```diff
@@ -345,6 +345,7 @@ if(NOT NO_OPENGL AND NOT ANDROID AND NOT IOS)
 endif()
 if(NOT NO_PTEX)
    find_package(PTex 2.0)
+   find_package(Zlib 1.2)
```
Comment (thomthom) [inline CMakeLists.txt]: Is there more needed to find Zlib on Windows? Where is it looking for Zlib?

Comment (manuelk) [inline CMakeLists.txt]: CMake ships with a FindZLIB module, but you have to set the root location... (-DZLIB_ROOT="...")

**Comment by davidgyu** [comment]
Thanks Manuel! This looks great. We'll pull this into dev now that 3.0.5 has been released. -David

**Pull request merged**